### PR TITLE
Revisit update scan ranges

### PR DESCRIPTION
**Pull request by Oscar-Pepper**
ontop of #1589 

- reworks the state module to allow for improved out-of-order sync logic
- adds chain tip scanning
- updates verification logic to match state changes

### CODE REVIEW

**Review by AloeareV — DISMISSED** [review]
Once again, approved contingent on the PR this is based atop landing!

**Review by AloeareV — DISMISSED** [review]

**Review by AloeareV — DISMISSED** [review]

**Review by zancas — APPROVED** [review]

**Pull request merged**
